sources/updater/updog/src/main.rs

```diff
@@ -453,6 +456,49 @@ fn initiate_reboot() -> Result<()> {
     Ok(())
 }
 
+/// Our underlying HTTP client, reqwest, supports proxies by reading the HTTPS_PROXY and NO_PROXY
+/// environment variables. Bottlerocket services can source proxy.env before running, but updog is
+/// not a service, so we read these values from the config file and add them to the environment
+/// here.
+fn set_https_proxy_environment_variables(
+    https_proxy: &Option<String>,
+    no_proxy: &Option<Vec<String>>,
+) -> Result<()> {
+    let proxy = match https_proxy {
+        Some(s) if !s.is_empty() => s.to_owned(),
+        // without https_proxy, no_proxy does nothing, so we are done
+        _ => return Ok(()),
+    };
+
+    // TODO - remove this if reqwest adds support for proxy addresses with no scheme
+    // reqwest needs a URL protocol scheme to be present, but other implementations assume
+    // `http://` as the scheme when none is present. We need to check the `HTTPS_PROXY` env and
+    // reset it with `http://` when no scheme is present
+    let proxy = if Url::parse(&proxy).is_ok() {
+        // the proxy value is OK, it has a scheme
+        debug!("setting HTTPS_PROXY={}", proxy);
+        proxy
+    } else {
+        // try prepending the default scheme
+        let prepended = format!("http://{}", proxy);
+        // now we expect a valid URL
+        let _ = Url::parse(&prepended).context(error::Proxy { proxy })?;
+        // now we know we have a string that will work with reqwest
+        debug!("prepended https:// and setting HTTPS_PROXY={}", prepended);
+        prepended
+    };
+
+    std::env::set_var("HTTPS_PROXY", &proxy);
+    if let Some(no_proxy) = no_proxy {
```

Comment:
Does reqwest also need a scheme for the no_proxy values?

Reply:
Testing has shown that protocol schemes are not needed for the NO_PROXY values.